File: src/main/java/edu/luc/comp433/api/payload/ProductRequest.java
package edu.luc.comp433.api.payload;

import edu.luc.comp433.business.dto.CategoryDTO;
import edu.luc.comp433.business.dto.ProductDTO;
import edu.luc.comp433.business.dto.SellerDTO;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;

import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import javax.xml.bind.annotation.XmlRootElement;

@Data
@NoArgsConstructor
@AllArgsConstructor
@XmlRootElement(name = "product")
public class ProductRequest {

    @NotNull
    @NonNull
    @Size(min = 3, max = 50)
    private String name;

    @Size(min = 10, max = 1000)
    private String description;

    @NotNull
    private Double listPrice;

    @NotNull
    private Long availableQuantity;

    @NotNull
    private Long sellerId;

    @NotNull
    private Long categoryId;

    public ProductDTO toDTO() {
        return new ProductDTO(null, name, description, listPrice, availableQuantity)
                .setCategoryDTO(new CategoryDTO(categoryId))
                .setSellerDTO(new SellerDTO(sellerId));
    }
}

File: front-end/api/product/ProductRequest.ts
import IRequest from "../IRequest";

export default class ProductRequest implements IRequest {
    private name: string;
    private description: string;
    private listPrice: number;
    private availableQuantity: number;
    private sellerId: number;
    private categoryId: number;


    constructor(name: string, description: string, listPrice: number, availableQuantity: number, sellerId: number, categoryId: number) {
        this.name = name;
        this.description = description;
        this.listPrice = listPrice;
        this.availableQuantity = availableQuantity;
        this.sellerId = sellerId;
        this.categoryId = categoryId;
    }
}